### PR TITLE
refactor docs, please check

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
- README has only the most important infos that a new reader needs in his first minute
  of contact with the project (to decide whether it is interesting or not)
- CHANGES shall later be a curated change log (== important changes between releases)
- separate docs into intro, using, project
- intro docs = include README, CHANGES (avoid duplication)

### CODE REVIEW

**Comment by jdkasten** [comment]
I think this is a great improvement!

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Documentation: https://letsencrypt.readthedocs.org/
+
+Software project: https://github.com/letsencrypt/lets-encrypt-preview
+
+Main Website: https://letsencrypt.org/
+
+IRC Channel: #letsencrypt on `Freenode`_
+
+Mailing list: `client-dev`_ (to subscribe without a Google account, send an
```
Comment (kuba) [inline README.rst]: line above gets rendered (in GitHub) as ***strong italics***, line below does not

File: docs/project.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Most importantly, **be consistent with the rest of the code**, please.
+
+1. Read [PEP 8 - Style Guide for Python Code]
+(https://www.python.org/dev/peps/pep-0008).
+
+2. Follow [Google Python Style Guide]
+(https://google-styleguide.googlecode.com/svn/trunk/pyguide.html),
+with the exception that we use [Sphinx](http://sphinx-doc.org/)-style
+documentation:
```
Comment (kuba) [inline docs/project.rst]: Links in 1. and 2. do not render well in GitHub, check preview.

File: docs/project.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+In order to generate the Sphinx documentation, run the following commands.
+
+::
+
+    ./venv/bin/python setup.py docs
+    cd docs
+    make clean html SPHINXBUILD=../venv/bin/sphinx-build
+
+
+This should generate documentation in the `docs/_build/html` directory.
```
Comment (kuba) [inline docs/project.rst]: backtics don't change rendering in GitHub

File: docs/using.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+::
+
+    sudo apt-get install python python-setuptools python-virtualenv python-dev \
+                 gcc swig dialog libaugeas0 libssl-dev ca-certificates
+
+
+Mac OSX
+-------
+
+::
```
Comment (kuba) [inline docs/using.rst]: It doesn't render correctly in GitHub. I believe you're missing an empty line here.

**Comment by kuba** [comment]
From the technical point of view, some of the things do not get rendered properly in GitHub viewer. I wonder if it's GitHub that fails with reStructuredText or if the documentation needs to be improved.

**Comment by ThomasWaldmann** [comment]
ok, PR updated. I hope I fixed all the stuff.

about viewing with github viewer: while it would be nice if it worked, i don't think this is critical as we point users to the rendered docs on readthedocs. thus rather correct sphinx output is important.

the only exception is the README (which is rendered automatically below the github repo view), which must render ok for both ways.

**Comment by jdkasten** [comment]
I made a slight change locally to the intro paragraph, but I am ready to pull this in once #202 happens.  Thanks @ThomasWaldmann.

**Comment by kuba** [comment]
@jdkasten Autodoc generation doesn't seem to be enough of a reason to block this one IMHO. README hasn't been updated that often and in those rare cases you can always build docs manually, by pressing a button. I guess we will have #202 working by the time we face that problem... ;)

**Comment by ThomasWaldmann** [comment]
yup, what @kuba said. :)

**Comment by jdkasten** [comment]
Basically our stance goes from generate the documentation yourself (which is guaranteed to be accurate) -> don't worry about it... we have everything online (which has been outdated several times).  Until I can get authorization for the various hooks I will just manually make sure that everything is in sync.

**Pull request merged**